modules/adfs/lib/IdP/ADFS.php

```diff
@@ -70,7 +71,13 @@ public static function ADFS_GenerateResponse($issuer, $target, $nameid, $attribu
 		foreach ($attributes as $name => $values) {
 			if ((!is_array($values)) || (count($values) == 0)) continue;
 			$hasValue = FALSE;
-			$r = '<saml:Attribute AttributeNamespace="http://schemas.xmlsoap.org/claims" AttributeName="' . htmlspecialchars($name) .'">';
+			if (strpos($name, '/')) {
+				$r = '<saml:Attribute AttributeNamespace="'.substr($name, 0, strrpos($name, '/', -1)).
+					'" AttributeName="'.substr($name, strrpos($name, '/', -1) + 1).'">';
```

Comment:
Here I think we should also use `htmlspecialchars()`, as in the other branch of the `if` clause.